Python/optimizer.c

```diff
@@ -731,7 +728,6 @@ translate_bytecode_to_trace(
                                     2 * INSTR_IP(instr, code));
                                 goto top;
                             }
-                            ADD_TO_TRACE(SAVE_IP, 0, 0);
```

Comment:
I had to look twice before understanding how we even get here. :-) I think we could use a comment here explaining that here `func == NULL`. (And/or a `DPRINT()` call?)